### PR TITLE
ATS-671: Split engines into fat & skinny modules (ATS-674)

### PR DESCRIPTION
**Pull request by eknizat**
* Transform engines produce a (Spring Boot JAR)
* Transform engines produce a stripped down skinny JAR
* All functionality remains the same

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
 
     - name: "Release"
       stage: release
-      if: commit_message ~= /\[trigger release\]/ AND branch ~= /^(master|SP\/.+|HF\/.+)$/
+      if: commit_message ~= /\[trigger release\]/ AND branch ~= /^(ATS-671_skinny_jars_in_modules|master|SP\/.+|HF\/.+)$/
```
Comment (montgolfiere) [inline .travis.yml]: I assume we revert this change (ie. should not be merged) ?

File: _ci/release.sh [hunk 2]
```diff
@@ -18,6 +18,8 @@ mvn -B \
     ${DRY_RUN} \
     -Prelease \
     "-Darguments=-Prelease -DskipTests -Dmaven.javadoc.skip -Dadditionalparam=-Xdoclint:none" \
+    -DreleaseVersion="2.1.2-test-0" \
+    -DdevelopmentVersion="2.1.2-SNAPSHOT" \
```
Comment (montgolfiere) [inline _ci/release.sh]: I assume we revert this change (ie. should not be merged) ?

Comment (montgolfiere) [reply _ci/release.sh]: We should also update pom.xml to 2.2.0-SNAPSHOT, eg.

mvn versions:set -DnewVersion=2.2.0-SNAPSHOT

File: _ci/static_analysis.sh [hunk 3]
```diff
@@ -13,11 +13,16 @@ java -jar vosp-api-wrappers-java-$VERACODE_WRAPPER_VERSION.jar -vid $VERACODE_AP
...
+     alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/target/alfresco-transform-pdf-renderer-*.jar \
+     alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer-boot/target/alfresco-transform-pdf-renderer-*.jar \
+     alfresco-transform-imagemagick/alfresco-transform-imagemagick/target/alfresco-transform-imagemagick-*.jar \
+     alfresco-transform-imagemagick/alfresco-transform-imagemagick-spring/target/alfresco-transform-imagemagick-spring-*.jar \
+     alfresco-transform-libreoffice/alfresco-transform-libreoffice/target/alfresco-transform-libreoffice-*.jar \
+     alfresco-transform-libreoffice/alfresco-transform-libreoffice-spring/target/alfresco-transform-libreoffice-spring-*.jar \
+     alfresco-transform-tika/alfresco-transform-tika/target/alfresco-transform-tika-*.jar \
+     alfresco-transform-tika/alfresco-transform-tika-spring/target/alfresco-transform-tika-spring-*.jar \
+     alfresco-transform-misc/alfresco-transform-misc/target/alfresco-transform-misc-*.jar \
+     alfresco-transform-misc/alfresco-transform-misc-spring/target/alfresco-transform-misc-spring*.jar \
```
Comment (montgolfiere) [inline _ci/static_analysis.sh]: Should we decide on either "-boot" (or "-spring") for consistency ?

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick-spring/pom.xml [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
-    <artifactId>alfresco-docker-imagemagick</artifactId>
-    <name>Alfresco Docker ImageMagick</name>
+    <artifactId>alfresco-transform-imagemagick-spring</artifactId>
+    <name>Alfresco Transform ImageMagick Spring Boot</name>
     <packaging>jar</packaging>
 
     <parent>
         <groupId>org.alfresco</groupId>
         <artifactId>alfresco-transform-core</artifactId>
         <version>2.1.2-SNAPSHOT</version>
```
Comment (montgolfiere) [inline alfresco-transform-imagemagick/alfresco-transform-imagemagick-spring/pom.xml]: We update the pom.xml (s) to 2.2.0-SNAPSHOT, eg.

mvn versions:set -DnewVersion=2.2.0-SNAPSHOT

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick/pom.xml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>alfresco-transform-imagemagick</artifactId>
+    <name>Alfresco Transform ImageMagick</name>
+    <packaging>jar</packaging>
+
+    <parent>
+        <artifactId>alfresco-transform-core</artifactId>
+        <groupId>org.alfresco</groupId>
+        <version>2.1.2-SNAPSHOT</version>
```
Comment (montgolfiere) [inline alfresco-transform-imagemagick/alfresco-transform-imagemagick/pom.xml]: Same comment as above ☝️

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick/pom.xml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.alfresco</groupId>
+            <artifactId>alfresco-transformer-base</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <!--<dependency>-->
+            <!--<groupId>org.dom4j</groupId>-->
+            <!--<artifactId>dom4j</artifactId>-->
+        <!--</dependency>-->
```
Comment (montgolfiere) [inline alfresco-transform-imagemagick/alfresco-transform-imagemagick/pom.xml]: Trivial: outside the scope of this JIRA but I wonder if we should just take the opportunity to clean-up and delete this commented out config ?

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/resources/engine_config.json [hunk 7]
```diff
@@ -0,0 +1,1158 @@
+{
```
Comment (montgolfiere) [inline alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/resources/engine_config.json]: I wonder why this appears as a diff in the PR rather than a rename ?

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/JodConverterSharedInstance.java [hunk 8]
```diff
@@ -26,22 +26,21 @@
  */
 package org.alfresco.transformer.executors;
 
-import static java.util.Arrays.asList;
-import static java.util.Objects.requireNonNull;
+import org.artofsolving.jodconverter.office.DefaultOfficeManagerConfiguration;
```
Comment (montgolfiere) [inline alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/JodConverterSharedInstance.java]: Minor: format only change ? Do we need team standard for IDE imports to reduce flip-flop when editing files ?

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java [hunk 9]
```diff
@@ -127,6 +121,10 @@ public void call(File sourceFile, File targetFile, String... args)
 
     public void convert(File sourceFile, File targetFile)
     {
+        if (jodconverter == null)
+        {
+            System.out.println("NULL!!!!");
+        }
```
Comment (montgolfiere) [inline alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java]: TODO - should use logger with a clear message and not drop through !!

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/resources/engine_config.json [hunk 10]
```diff
@@ -0,0 +1,235 @@
+{
```
Comment (montgolfiere) [inline alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/resources/engine_config.json]: Same comment as above for earlier engine_config.json ☝️

File: alfresco-transform-misc/alfresco-transform-misc-spring/src/test/java/org/alfresco/transformer/MiscControllerTest.java [hunk 11]
```diff
@@ -62,7 +62,6 @@
 
 @RunWith(SpringRunner.class)
 @WebMvcTest(MiscController.class)
-@Import({SelectingTransformer.class})
```
Comment (montgolfiere) [inline alfresco-transform-misc/alfresco-transform-misc-spring/src/test/java/org/alfresco/transformer/MiscControllerTest.java]: I assume this was an intentional removal ?

**Comment by montgolfiere** [comment]
Quick note for to any reviewers. 

We plan to merge this to master this week (eg. 2.2.0-SNAPSHOT). However, we may tweak folder/artefact names further (prior to an initial alpha 2.2.0-A1 release) as we move on to create the "all-in-one" Core T-Engine that combines the existing five T-Engines (ATS-675).

File: _ci/release.sh (hunk 2 above)
Comment (montgolfiere) [reply _ci/release.sh]: Thanks @PunchedChimera for fixing here: https://github.com/Alfresco/alfresco-transform-core/pull/192/commits/1b95fa367713468a8287dead1d5295d5b7bd6374

File: .travis.yml (hunk 1 above)
Comment (PunchedChimera) [reply .travis.yml]: Removed

File: _ci/release.sh (hunk 2 above)
Comment (PunchedChimera) [reply _ci/release.sh]: Removed and versions updated to 2.2.0-SNAPSHOT

File: _ci/static_analysis.sh (hunk 3 above)
Comment (PunchedChimera) [reply _ci/static_analysis.sh]: Renamed to "-boot"

File: alfresco-transform-misc/alfresco-transform-misc-spring/src/test/java/org/alfresco/transformer/MiscControllerTest.java (hunk 11 above)
Comment (PunchedChimera) [reply alfresco-transform-misc/alfresco-transform-misc-spring/src/test/java/org/alfresco/transformer/MiscControllerTest.java]: @eknizat

File: alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/resources/engine_config.json (hunk 7 above)
Comment (PunchedChimera) [reply alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/resources/engine_config.json]: It's because the file location has changed slightly. Previously it was in `.../src/resources/` it is now in `.../src/main/resources/`

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java (hunk 9 above)
Comment (PunchedChimera) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/LibreOfficeJavaExecutor.java]: Agreed, improving this area of code is out of scope of the PR as add additional checks or erroring may affect functionality.

File: alfresco-transform-misc/alfresco-transform-misc-spring/src/test/java/org/alfresco/transformer/MiscControllerTest.java (hunk 11 above)
Comment (eknizat) [reply alfresco-transform-misc/alfresco-transform-misc-spring/src/test/java/org/alfresco/transformer/MiscControllerTest.java]: Yes, intentional

File: alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/JodConverterSharedInstance.java (hunk 8 above)
Comment (PunchedChimera) [reply alfresco-transform-libreoffice/alfresco-transform-libreoffice/src/main/java/org/alfresco/transformer/executors/JodConverterSharedInstance.java]: Agreed, changes here are not needed. Reverted to original.

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/main/java/org/alfresco/transformer/Application.java [hunk 12]
```diff
@@ -1,74 +1,74 @@
...
-        logger.info("--------------------------------------------------------------------------------------------------------------------------------------------------------------");
-        logger.info("The transformers in this project use libraries from Apache. See the license at http://www.apache.org/licenses/LICENSE-2.0. or in /Apache\\\\ 2.0.txt");
-        logger.info("Additional libraries used:");
-        logger.info("* htmlparser http://htmlparser.sourceforge.net/license.html");
-        logger.info("--------------------------------------------------------------------------------------------------------------------------------------------------------------");
-
-        logger.info("Starting application components... Done");
-    }
-}
+/*
```
Comment (montgolfiere) [inline alfresco-transform-misc/alfresco-transform-misc-boot/src/main/java/org/alfresco/transformer/Application.java]: Format only changes should be done in separate PRs - did anything else change other than line-endings (cc @eknizat @PunchedChimera ) ?

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/main/java/org/alfresco/transformer/MiscController.java [hunk 13]
```diff
@@ -1,159 +1,158 @@
-/*
```
Comment (montgolfiere) [inline alfresco-transform-misc/alfresco-transform-misc-boot/src/main/java/org/alfresco/transformer/MiscController.java]: Format only changes should be done in separate PRs - did anything else change other than line-endings (cc @eknizat @PunchedChimera ) ?

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscQueueTransformServiceIT.java [hunk 14]
```diff
@@ -1,58 +1,58 @@
-/*
```
Comment (montgolfiere) [inline alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscQueueTransformServiceIT.java]: Format only changes should be done in separate PRs - did anything else change other than line-endings (cc @eknizat @PunchedChimera ) ?

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java [hunk 15]
```diff
@@ -1,48 +1,48 @@
-/*
```
Comment (montgolfiere) [inline alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java]: Format only changes should be done in separate PRs - did anything else change other than line-endings (cc @eknizat @PunchedChimera ) ?

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/resources/quick.alternative.eml [hunk 16]
```diff
@@ -1,30 +1,30 @@
-MIME-Version: 1.0
```
Comment (montgolfiere) [inline alfresco-transform-misc/alfresco-transform-misc-boot/src/test/resources/quick.alternative.eml]: Format only changes should be done in separate PRs - did anything else change other than line-endings (cc @eknizat @PunchedChimera ) ?

File: alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/LICENSES.md [hunk 17]
```diff
@@ -0,0 +1,4 @@
+### Licenses
```
Comment (montgolfiere) [inline alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/LICENSES.md]: File rename/move ? Same as engine_config.json earlier ... ?

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java (hunk 15 above)
Comment (PunchedChimera) [reply alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java]: There are no changes.

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/resources/quick.alternative.eml (hunk 16 above)
Comment (PunchedChimera) [reply alfresco-transform-misc/alfresco-transform-misc-boot/src/test/resources/quick.alternative.eml]: There are no changes

File: alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/LICENSES.md (hunk 17 above)
Comment (PunchedChimera) [reply alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/LICENSES.md]: Yes, as before. No changes have been made to this file.

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java (hunk 15 above)
Comment (montgolfiere) [reply alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java]: If no changes then any idea why it is appearing as a diff in this PR ? 

Eric also mentioned that when "misc" was originally added it used CRLF unlike rest of project that uses just LF. 

If so, we could do a sweep (preferably as a separate PR) to make this consistent and reduce likelihood of format-only changes due to git / ide settings ?

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/main/java/org/alfresco/transformer/Application.java (hunk 12 above)
Comment (PunchedChimera) [reply alfresco-transform-misc/alfresco-transform-misc-boot/src/main/java/org/alfresco/transformer/Application.java]: No changes to this file.

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java (hunk 15 above)
Comment (PunchedChimera) [reply alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java]: I've noticed that my git has the following config: `core.autocrlf=true`
If there were areas using LF then this is why we are seeing the changes.

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscControllerTest.java [hunk 18]
```diff
@@ -1,431 +1,430 @@
...
-import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
-import org.springframework.context.annotation.Import;
-import org.springframework.mock.web.MockMultipartFile;
-import org.springframework.test.context.junit4.SpringRunner;
-import org.springframework.test.web.servlet.MvcResult;
-import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
-
-@RunWith(SpringRunner.class)
-@WebMvcTest(MiscController.class)
-@Import({SelectingTransformer.class})
```
Comment (PunchedChimera) [inline alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscControllerTest.java]: This line is removed `@Import({SelectingTransformer.class})`

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/main/java/org/alfresco/transformer/MiscController.java [hunk 19]
```diff
@@ -1,159 +1,158 @@
...
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.multipart.MultipartFile;
-
-@Controller
-public class MiscController extends AbstractTransformerController
-{
-    private static final Logger logger = LoggerFactory.getLogger(MiscController.class);
-
-    @Autowired
-    private SelectingTransformer transformer;
```
Comment (PunchedChimera) [inline alfresco-transform-misc/alfresco-transform-misc-boot/src/main/java/org/alfresco/transformer/MiscController.java]: Change was made here. These lines changed to: `private SelectingTransformer transformer = new SelectingTransformer();`

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java (hunk 15 above)
Comment (montgolfiere) [reply alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java]: OK, thanks. Will you fix now (as part of this PR) or later to be consistent with rest of project ?

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/main/java/org/alfresco/transformer/Application.java (hunk 12 above)
Comment (PunchedChimera) [reply alfresco-transform-misc/alfresco-transform-misc-boot/src/main/java/org/alfresco/transformer/Application.java]: line changes fixed now set to CRLF

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java (hunk 15 above)
Comment (PunchedChimera) [reply alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscTransformerHttpRequestTest.java]: line changes fixed now set to CRLF

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/resources/quick.alternative.eml (hunk 16 above)
Comment (PunchedChimera) [reply alfresco-transform-misc/alfresco-transform-misc-boot/src/test/resources/quick.alternative.eml]: line changes fixed now set to CRLF

File: alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscControllerTest.java (hunk 18 above)
Comment (PunchedChimera) [reply alfresco-transform-misc/alfresco-transform-misc-boot/src/test/java/org/alfresco/transformer/MiscControllerTest.java]: line changes fixed now set to CRLF Should be easier to see change

**Comment by PunchedChimera** [comment]
I can see that the "Spring Boot" modules also have the `engine_config.json`
(For example: `alfresco-transform-*\alfresco-transform-*-boot\src\main\resources\engine_config.json`)
Should these still exist in the "Spring Boot" modules, my understanding was that these should only be in the "skinny" jars? @eknizat @montgolfiere 
This is why there are currently `engine_config.json` files appearing in the PR as new files. Because there are now duplicated across the "skinny" and "fat" modules @montgolfiere

**Comment by montgolfiere** [comment]
I recommend smaller steps. Let's remove the duplicates if they're not used yet (ie. remove from the JARs ?). We can adjust A/Cs for ATS-674 and then revisit in next set of stories ?

File: alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/LICENSES.md (hunk 17 above)
Comment (montgolfiere) [reply alfresco-transform-pdf-renderer/alfresco-transform-pdf-renderer/LICENSES.md]: I assume line-endings only ?
